jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/MethodScope.java

```diff
@@ -171,11 +177,49 @@ public boolean isGetter() {
     public <A extends Annotation> A getAnnotation(Class<A> annotationClass) {
         A annotation = super.getAnnotation(annotationClass);
         if (annotation == null) {
-            annotation = this.getRawMember().getAnnotatedReturnType().getAnnotation(annotationClass);
+            annotation = findGetterCandidates(this.getRawMember()).stream().map(m -> {
+                return Optional.ofNullable(m.getAnnotation(annotationClass))
+                        .orElse(m.getAnnotatedReturnType().getAnnotation(annotationClass));
+            }).filter(Objects::nonNull).findFirst().orElse(null);
         }
         return annotation;
     }
 
+    private Collection<Method> findGetterCandidates(Method m) {
```

Comment:
**suggestion**: This method is only invoked once with `this.getRawMember()` as its single parameter value.
Therefore, no need to provide any parameter here.